vendor_data.py

```diff
@@ -0,0 +1,156 @@
+            VendorName should be a string, if not a TypeError is raised.
+            TargLang should be a string, if not a TypeError is raised.
+            WordRate should be a float, if not a TypeError is raised.
+            WordRate should NOT be > 0.15, if not a ValueError is raised.
+            VendorMail should be a string, if not a TypeError is raised.
+            VendorMail should be an email, if not a ValueError is raised.
+            CatTool should be a string, if not a TypeError is raised.
+            CatTool should be one of the options in the list "CatTools", if not a ValueError is raised.
+        """
+        
+        if type(VendorName) !=str:
+            raise TypeError("VendorName should be a string!")
+        if type(TargLang) != str:
+            raise TypeError("TargLang should be a string!")
+        if type(WordRate) != float:
+            raise TypeError("WordRate should be a float!")
+        if WordRate:
+            if WordRate > 0.15:
+                raise ValueError("This vendor is too expensive, consider another one.")
+        elif WordRate == 0.00:
+            raise ValueError("Word rate cannot be 0.00.")
+        else:
+            self.WordRate = WordRate
+        if type(CatTool) != str:
+            raise TypeError("CatTool should be a string!")
+        if type(VendorMail) != str:
+            raise TypeError ("VendorMail should be a string!")
+        if VendorMail:   
+            if(re.search(regex_mail,VendorMail)):
+                self.VendorMail = VendorMail
+            else:
+                raise ValueError("Please insert a valid email address.")
+        else:
+            self.VendorMail = VendorMail
+        if CatTool in VendorData.CatTools:
+            self.CatTool = CatTool
+        else:
+            raise ValueError("This CAT tool is not valid. Run 'VendorData.CatTools' to check options.")
+            
+    def set_mail(self,mail):
+        regex_mail = "^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$" #regex used to validate email
+        if (re.search(regex_mail,mail)):
+            self.VendorMail = mail
+        else:
+            raise ValueError("Please insert a valid email address.")      
+   
+    def to_excel(self):
+        data = {
+            "Target Language": [self.TargLang],
+            "Vendor": [self.VendorName],
+            "E-mail": [self.VendorMail],
+            "Word rate": [self.WordRate],
+            "CAT-tool": [self.CatTool]
+        }
+        filename = "_".join([str(self.ProjectName), str(self.SourceLang)])
+        df = pd.DataFrame(data)
+        if os.path.exists(f"{filename}.xlsx") != True:
+            df.to_excel(f"{filename}.xlsx", index=False, sheet_name="VendorData")
+        else:
+            with pd.ExcelWriter(f"{filename}.xlsx", mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
+                df.to_excel(writer, sheet_name = "VendorData", startrow=writer.sheets["VendorData"].max_row, header = None, index=False)

```

Comment:
Looks good!
I would split lines 109 and 110 in different lines like you did with the dictionary right above.
And I would approach this mergin in a different, safer way. (To talk in person)